shared/rakefiles/entrypoint.rake

```diff
@@ -122,12 +111,11 @@ end
 desc '[ADVANCED] Run arbitrary exekube command -- rake sh["kubectl --namespace gpii get pods"]'
 task :sh, [:cmd] => [:set_vars] do |taskname, args|
   if args[:cmd]
-    cmd = args[:cmd]
+    sh "#{@exekube_cmd} rake xk['#{args[:cmd]}',skip_secret_mgmt]"
   else
     puts "Argument :cmd -- the command to run inside the exekube container -- not present, defaulting to bash"
-    cmd = "bash"
+    sh "#{@exekube_cmd} sh"
```

Comment:
@amatas Why do we need these changes? `sh "#{@exekube_cmd} sh"` is not the same  as `sh "#{@exekube_cmd} rake xk['#{cmd}',skip_secret_mgmt]"`. With this, secrets will not be applied when you call `rake sh` without providing command, and shell output will not be obfuscated.

Reply:
I introduced this changes in order to avoid load functions if I only need a shell to debug inside the container. In which cases do we need to load the secrets when we invoke "rake sh"?

Before these changes, if I just wanted to test my TF code I had to authenticate against gcloud and do some stuff to get the SA, which in that time when I was debugging they didn't exist at GCP.

Reply:
@amatas Ok, I understand this now. Example of situation when we need secrets inside exekube shell is [CouchDB data restoration](https://github.com/gpii-ops/gpii-infra/tree/master/gcp#restoring-couchdb-data). Different behaviors for `rake sh` that simply invokes bash and for `rake sh[bash]` that does the same but via ruby code in container, is a bit confusing. Maybe introduce something like `:plain_sh` for that purpose? But I am okay leaving this as it is.

Reply:
I would prefer to have secrets by default - you need these to be able to debug or work at all with TF state, without you can't do much (well anything with TF)

Reply:
I've leave the original task as `:rake_shell` and created a new one `:plain_shell` for covering those cases where someone wants to get into a shell without external dependencies